Sources/BBCodeParser.php

```diff
@@ -4147,7 +4151,7 @@ protected function transformToHtml(array $tag, array $params): void
 
 			$data = [
 				substr($this->message, $pos2 + ($quoted == false ? 1 : 7), $pos3 - ($pos2 + ($quoted == false ? 1 : 7))),
-				substr($this->message, $this->pos1, $pos2 - $this->pos1),
+				substr($this->message, $this->pos1, (int) ($pos2 - $this->pos1)),
```

Comment:
Casting the result of the expression to int is unnecessary. The result of the arithmetic operation will already be an integer.